nodestream/project/plugin.py

```diff
@@ -1,21 +1,27 @@
-from dataclasses import dataclass
-from typing import Set
+from dataclasses import dataclass, field
+from importlib import resources
+from typing import Dict
+
+from schema import Or
 
 from ..file_io import LoadsFromYamlFile
 from ..pipeline.scope_config import ScopeConfig
+from ..project.pipeline_definition import PipelineConfiguration, PipelineDefinition
+from ..project.pipeline_scope import PipelineScope
 
 
 @dataclass
 class PluginConfiguration(LoadsFromYamlFile):
-    """A `PluginScope` represents a collection of configuration for a plugin.
+    """A `PluginConfiguration` represents a collection of configuration for a plugin.
 
     A config is a key value pair object in a nodestream scope including plugins.
     It contains a collection of configuration key value pairs to be used by the pipelines of a scope.
     """
 
     name: str
+    pipelines_by_name: Dict[str, PipelineDefinition] = field(default_factory=dict)
     config: ScopeConfig = None
-    targets: Set[str] = None
+    pipeline_configuration: PipelineConfiguration = None
```

Comment:
I think this field _is_ required. So it should probably move above `config` and remove the `= None` part.

Reply:
Its not required, last line of this file